File: 7주차/이연걸/크게만들기.py
N, K = map(int, input().split())
num = list(input().rstrip())
stack = []

for i in range(N):
    while K > 0 and stack and stack[-1] < num[i]:
        stack.pop()
        K -= 1
    stack.append(num[i])

while K != 0:
    stack.pop()
    K -= 1

for s in stack:
    print(s, end='')

File: 7주차/임유정/크게 만들기.js
// https://www.acmicpc.net/problem/2812

function solution(n, k, num) {
  const stack = [];

  for (const curr of n) {
    if (stack.length && stack[stack.length - 1] < curr && k) {
      stack.pop();
      k--;
    }
    stack.push(curr);
  }
  return stack.join("").substring(0, stack.length - k);
}

solution(10, 4, 4177252841);
